### PR TITLE
add three-octree.d.ts and three-ctmloader.d.ts

### PR DESCRIPTION
**Pull request by omni360**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by florentpoujol** [comment]
👍

File: three/index.d.ts [hunk 1]
```diff
@@ -1996,7 +1998,7 @@ declare namespace THREE {
         mimeType: MimeType;
         path: string;
         responseType: string;
-        withCredentials: string        
+        withCredentials: string
```
Comment (aozgaa) [inline three/index.d.ts]: Can you add a `tslint.json` file to the directory? This would catch the missing semi-colon here.

Comment (omni360) [inline three/index.d.ts]: ok.3q!

**Comment by aozgaa** [comment]
It looks like you linted the directory but didn't actually add a `tslint.json` per https://github.com/DefinitelyTyped/DefinitelyTyped/#lint .

Could you add such a `tslint.json` at `three/tslint.json`?

Additionally, writing different commit messages can help ease readability of your changes.

**Comment by StephenRUK** [comment]
Hi folks! I'm not sure if this is the right place to comment, but I noticed some of the type files include  `<reference types="...">` statements. This causes compile errors using npm 3.10.8 and TypeScript 2.0.10, as e.g. "three" (same package) and "webvr-api" (not in use in our project) cannot be resolved.

Can these directives be omitted? Otherwise a workaround is to set `"noResolve": true` in tsconfig, which I doubt works for everyone.

Ref: https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html

**Pull request merged**